cmake/pip_utils.cmake

```diff
@@ -0,0 +1,172 @@
+endfunction()
+
+function(pip_configure_package_file_hierarchy)
+    foreach(file IN ITEMS ${ARGV})
+        if(IS_ABSOLUTE ${file})
+            message(AUTHOR_WARNING "Files for 'pip_configure_package_file_hierarchy' (here '${file}') should be relative to the package directory ('PYTHON_PACKAGE_DIR') in the same way for input and output")
+        endif()
+        pip_configure_package_file(${file} ${file})
+    endforeach()
+endfunction()
+
+function(pip_configure_subpackage_file in_file out_file)
+    configure_file(${in_file} ${PYTHON_PACKAGE_CONTENT_DIR}/${out_file})
+    list(APPEND PIP_CONFIGURE_OUTPUT_FILES ${PYTHON_PACKAGE_CONTENT_DIR}/${out_file})
+    set(PIP_CONFIGURE_OUTPUT_FILES ${PIP_CONFIGURE_OUTPUT_FILES} PARENT_SCOPE)
+endfunction()
+
+function(pip_add_so_target)
+    set(options)
+    set(oneValueArgs NAME SO_NAME)
+    set(multiValueArgs DEPENDS)
+    cmake_parse_arguments(ARGS "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    add_custom_command(
+        OUTPUT ${WORKING_DIR}/so.md5
+        DEPENDS ${ARGS_SO_NAME} ${ARGS_DEPENDS}
+        COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:${ARGS_SO_NAME}> ${PYTHON_PACKAGE_CONTENT_DIR}/$<TARGET_FILE_NAME:${ARGS_SO_NAME}>
+        COMMAND md5sum ${PYTHON_PACKAGE_CONTENT_DIR}/$<TARGET_FILE_NAME:${ARGS_SO_NAME}> > ${WORKING_DIR}/so.md5
+        WORKING_DIRECTORY ${PYTHON_PACKAGE_DIR}
+        VERBATIM
+    )
+    add_custom_target(
+        ${ARGS_NAME}-target
+        ALL
+        DEPENDS ${WORKING_DIR}/so.md5
+        VERBATIM
+    )
+    set(${ARGS_NAME} ${ARGS_NAME}-target ${WORKING_DIR}/so.md5 PARENT_SCOPE)
+endfunction()
+
+function(pip_add_html_target)
+    set(options)
+    set(oneValueArgs NAME ENABLE_HTML_DOC)
+    set(multiValueArgs DEPENDS)
+    cmake_parse_arguments(ARGS "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    # TODO: Add support for namespace packages
+    if(ARGS_ENABLE_HTML_DOC)
+        message(FATAL_ERROR "Not implemented yet")
+        # add_custom_command(
+        #         OUTPUT ${PYTHON_PACKAGE_DIR}/${PYTHON_LIB_NAME}.html
+        #         DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/so.md5 ${GENERATE_CONFIGURE_OUTPUT_FILES} python-opentera-webrtc-native-client-so
+        #         COMMAND /usr/bin/env python3 -m pydoc -w opentera.webrtc.native_client
+        #         COMMAND /usr/bin/env python3 -m pydoc -w opentera.webrtc.native_client.${PYTHON_LIB_NAME}
+        #         WORKING_DIRECTORY ${PYTHON_PACKAGE_DIR}
+        #         VERBATIM
+        #     )
+        #     add_custom_target(
+        #     python-opentera-webrtc-native-client-html
+        #     ALL
+        #     DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/${PYTHON_LIB_NAME}.html
+        #     VERBATIM
+        # )
+        # set(HTML_DOC_DEPENDENCY_OPTION python-opentera-webrtc-native-client-html ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/${PYTHON_LIB_NAME}.html)
+    else()
```

Comment:
We might need to used something more complete than `pydoc`, like maybe `pdoc3`

Reply:
Is `pdoc3` compatible with pybind11? 
Why is the command commented?

Reply:
This is the old implementation, it won't work currently

Not sure about that. I can't get it to work, so I would guess not.